### PR TITLE
dist/tools/desvirt: proper desvirt integration

### PR DESCRIPTION
**Pull request by cgundogan**
This PR integrates `desvirt`[1] into the RIOT make system.

E.g. for the `gnrc_networking` example:
```
TOPO_TYPE=line TOPO_SIZE=4 make desvirt-topology
TOPO=line4 make desvirt-define
make desvirt-list
TOPO=line4 make desvirt-start
nc localhost 4711
ping6 ff02::1
TOPO=line4 make desvirt-stop desvirt-undefine
make desvirt-list
make desvirt-clean (deletes all topology files)
```
[1] https://github.com/des-testbed/desvirt

~~depends on https://github.com/des-testbed/desvirt/pull/17~~
~~Do not merge before I remove the "DELME" commit.~~

### CODE REVIEW

File: dist/tools/desvirt/README.desvirt.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+https://github.com/des-testbed/desvirt
+
+## Control DESVIRT via Make
+
+### Variables
+
+This Makefile introduces some additional variables:
+ * TOPO
+ * TOPO_SIZE
+ * TOPO_TYPE
```
Comment (OlegHahm) [inline dist/tools/desvirt/README.desvirt.md]: An explanation for these variables is missing.

File: dist/tools/desvirt/Makefile.desvirt [hunk 2]
```diff
@@ -0,0 +1,57 @@
+TOOL_NAME=desvirt
+TOOL_URL=https://github.com/cgundogan/desvirt.git
```
Comment (OlegHahm) [inline dist/tools/desvirt/Makefile.desvirt]: Don't forget to revert this. ;)

File: dist/tools/desvirt/README.desvirt.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * desvirt-undefine
+ * desvirt-list
+ * desvirt-start
+ * desvirt-stop
+ * desvirt-clean
+ * desvirt-distclean
+
+#### desvirt-topology
+
+This target creates a new topology file in RIOTBASE/dist/tools/desvirt/desvirt/.desvirt
```
Comment (OlegHahm) [inline dist/tools/desvirt/README.desvirt.md]: Maybe add backticks around `RIOTBASE...desvirt`

**Comment by OlegHahm** [comment]
I think we should add a note on how to connect to these native instances via `pyterm -ts`.

**Comment by cgundogan** [comment]
>I think we should add a note on how to connect to these native instances via pyterm -ts.

you seem to know more than me on this topic (:
I have to play around with pyterm.

File: dist/tools/desvirt/README.desvirt.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+riotnative     : Start the RIOT: socat EXEC:'/home/pschmerzl/RIOT/examples/gnrc_networking/bin/native/gnrc_networking.elf line4_3',end-close,stderr,pty TCP-L:4713,reuseaddr,fork
+riotnative     : PID: 18237
+vm             : Defining RIOT native process line4_2
+riotnative     : Start the RIOT: socat EXEC:'/home/pschmerzl/RIOT/examples/gnrc_networking/bin/native/gnrc_networking.elf line4_2',end-close,stderr,pty TCP-L:4714,reuseaddr,fork
+riotnative     : PID: 18238
+
+```
+
+Once the topology is running, you can use `pyterm` to connect to the various
+instances, e.g. calling `pyterm -ts 4712` to connect to node `line4_0` from the
```
Comment (cgundogan) [inline dist/tools/desvirt/README.desvirt.md]: @OlegHahm  should we say that it is also possible to use `netcat` instead of `pyterm`?

**Comment by cgundogan** [comment]
addressed comments

**Comment by OlegHahm** [comment]
ACK

File: dist/tools/desvirt/README.desvirt.md (hunk 4 above)
Comment (cgundogan) [inline dist/tools/desvirt/README.desvirt.md]: @OlegHahm actually it's not that important .. and travis already built successfully (:

**Comment by cgundogan** [comment]
thanks for the quick review!

**Pull request merged**